packages/net-stubbing/lib/server/middleware/request.ts

```diff
@@ -92,8 +92,23 @@ export const InterceptRequest: RequestMiddleware = async function () {
         return resolve()
       }
 
+      const onClose = (): void => {
+        req.body = 'Request canceled before body could be determined.'
```

Comment:
without changes here, there are instances (some sort of race condition) where the pre-request message hangs and isn't sent up to the driver. Without the pre-request first being sent up errors don't have a 'request' to match to and are discarded

https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cy/net-stubbing/events/network-error.ts#L11-L13

In these instances concatStream never calls its callback and doesn't error. So req.body is never resolved. Thus if we get into that situation through either the close event or because the response is destroyed, I'm overriding body with the error message. We could skip changing the body message if you think it's appropriate, but i'd just be empty when maybe it shouldn't be?

Reply:
Since there's already an error attached to the request, I think there's no need to set an error message here.